test/SpotifyAPITest.cpp

```diff
@@ -0,0 +1,685 @@
+/**
+ * @file    SpotifyAPITest.cpp
+ * @author  Stefan Jahn <stefan.jahn332@gmail.com>
+ * @brief   Test Class for the SpotifyAPI Class
+ * @details for correct testing a valid access token must be set
+ * the functions getAccessToken and refreshAccessToken are excluded from the
+ * tests, because they can not be tested without a user interaction, or other
+ * secret data which shouldn't be pushed in public repos
+ */
+
+#include <gtest/gtest.h>
+
+#include "Spotify/SpotifyAPI.h"
+#include "json/json.hpp"
+
+using namespace std;
+using namespace SpotifyApi;
+
+class SpotifyAPITest : public ::testing::Test {
+ public:
+  virtual void SetUp() {
+    // check if access token is valid
+    auto testRet = mAPI.search(mAccessToken, "linkinPark", QueryType::track);
+
+    if (auto value = get_if<Error>(&testRet)) {
+      value = value;
+      // if we get an error here, the access token is not valid or returned some
+      // other error
+      cout << "CAUTION: Access Token is not valid, Tests get aborted.." << endl;
+      exit(-1);
+    }
+  }
```

Comment:
is it ok to handle this with a exit? or should we accept it if the testing fails when the access token has been expired?

Reply:
i think you can actually call `ASSERT_*` in `SetUp`, but that would mark the test as failed (do we want that?)
as an alternative we could use [`GTEST_SKIP`](https://stackoverflow.com/a/53448080/4440121)

Reply:
The skip looks not so bad, will try it tomorrow